lambda/translcore.ml

```diff
@@ -471,14 +471,21 @@ and transl_exp0 ~in_new_scope ~scopes e =
                   event_before ~scopes ifso (transl_exp ~scopes ifso),
                   lambda_unit)
   | Texp_sequence(expr1, expr2) ->
-      Lsequence(transl_exp ~scopes expr1,
+      let lam1 = transl_exp ~scopes expr1 in
+      let mode1 = transl_value_mode expr1.exp_mode in
+      let lam1 = maybe_region Alloc_heap [mode1] lam1 in
+      Lsequence(lam1,
                 event_before ~scopes expr2 (transl_exp ~scopes expr2))
   | Texp_while(cond, body) ->
-      Lwhile(transl_exp ~scopes cond,
-             event_before ~scopes body (transl_exp ~scopes body))
+      let body_lam = event_before ~scopes body (transl_exp ~scopes body) in
+      let body_mode = transl_value_mode body.exp_mode in
+      let body_lam = maybe_region Alloc_heap [body_mode] body_lam in
+      Lwhile(transl_exp ~scopes cond, body_lam)
```

Comment:
Couldn't this just be `Lregion body_lam`? It's going to effort to ensure that `maybe_region` unconditionally inserts `Lregion`.

Reply:
I don't think it will be unconditionally inserted. It will only be inserted if `body_mode` is `Alloc_local`.

Reply:
Isn't it always Alloc_local? The body is typechecked by `type_statement` that uses `alloc_local` as the mode.

Reply:
Ah, right. I need to change the typing there, and probably elsewhere to use variables instead of `alloc_local` in such places. Once #37 lands that will make this be almost always `Alloc_heap` instead of `Alloc_local`.